src/enemy.rs

```diff
@@ -0,0 +1,287 @@
+use std::collections::HashMap;
+use bevy::prelude::*;
+use crate::player::*;
+use std::time::Duration;
+use bevy::time::TimerMode::Repeating;
+use rand::{Rng, thread_rng};
+use crate::steering::{SteeringBundle, SteeringHost, SteerSeek};
+use crate::stats::*;
+use serde_json;
+use serde::{Deserialize, Serialize};
+
+pub struct EnemyPlugin;
+
+impl EnemyBundle {
+    fn new(h: Health, d: Damage, r: Rewards) -> Self {
+        Self {
+            enemy: Enemy,
+            health: h,
+            damage: d,
+            rewards: r,
+        }
+    }
+}
+/*
+todolist
+Walk towards player [V]
+Damage player when almost in the player (atm dying).
+Proper spawn of multiple enemies [V]
+Die. [V]
+ */
+impl Plugin for EnemyPlugin {
+    fn build(&self, app: &mut App) {
+        app
+            .insert_resource(CurrentWave{ num: 1, timer: Timer::new(Duration::from_secs(30, ),Repeating), need_wave_spawn: true })
+            .insert_resource(GlobalTimeTickerResource(Timer::new(Duration::from_secs(5), TimerMode::Repeating)))
+            .add_systems(Startup, enemy_factory)
+            .add_systems(Update, (
+                spawn,
+                movement,
+                check_health,
+                change_wave,
+                global_timer_tick
+            ));
+    }
+}
+
+#[derive(Resource)]
+struct GlobalTimeTickerResource(Timer);
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SpawnWave {
+    from: u16,
+    to: u16,
+    spawn_time: u64,
+}
+
+#[derive(Component, Clone, Debug)]
+struct Rewards {
+    exp: u64,
+    items: &'static str, // TODO: When <Item> class is implemented, remove this mock up
+}
+
+#[derive(Component, Clone, Debug)]
+pub struct Enemy;
+
+#[derive(Bundle)]
+struct EnemyBundle {
+    enemy: Enemy,
+    health: Health,
+    damage: Damage,
+    rewards: Rewards,
+}
+
+#[derive(Component, Clone, Debug)]
+struct EnemySpawnComponent {
+    name: String,
+    enemy: Enemy,
+    health: Health,
+    damage: Damage,
+    max_velocity: MaxVelocity,
+    max_force: MaxForce,
+    mass: Mass,
+    texture: Handle<Image>,
+    rewards: Rewards,
+    is_elite: Option<bool>,
+    timer: Timer,
+}
+
+#[derive(Resource, Debug, Clone)]
+struct EnemySpawners(HashMap<u16, Vec<EnemySpawnComponent>>);
+
+impl EnemySpawnComponent {
+    fn new(name: String, enemy: Enemy, health: Health, damage: Damage, max_velocity: MaxVelocity, max_force: MaxForce, mass: Mass, texture: Handle<Image>,is_elite: Option<bool>, rewards: Rewards) -> Self {
```

Comment:
You don't need a constructor here, you can just use struct initialization syntax, like:
```rust
EnemySpawnComponent {
    name: "my name".to_string(),
    enemy: Enemy::default(),
    ...
};
```

Reply:
done